docs/content/Getting Started.fsx

```diff
@@ -5,217 +5,124 @@ categoryindex: 1
-</PropertyGroup>
-
-<Target Name="_AddAnalyzersToOutput">
+<Project>
+
     <ItemGroup>
-        <TfmSpecificPackageFile Include="$(OutputPath)\$(AssemblyName).dll" PackagePath="analyzers/dotnet/fs" />
+        <ProjectsToAnalyze Include="src\**\*.fsproj" />
     </ItemGroup>
-</Target>
-```
-
-`SuppressDependenciesWhenPacking` will prevent NuGet from taking your analyzer dependencies into account when it is being installed in a project. This most common use-case here would be `FSharp.Core`. 
-
-`TargetsForTfmSpecificContentInPackage` will make sure the `_AddAnalyzersToOutput` runs to copy the assembly into the `analyzers/dotnet/fs` folder.
 
-`IncludeBuildOutput` will ensure your `*.nupkg` doesn't have a `lib` folder.
+    <Target Name="AnalyzeSolution">
+        <Exec Command="dotnet build -c Release $(SolutionFileName)" />
+        <MSBuild
+                Projects="@(ProjectsToAnalyze)"
+                Targets="AnalyzeProject"
+                Properties="DesignTimeBuild=True;Configuration=Release;ProvideCommandLineArgs=True;SkipCompilerExecution=True" />
+    </Target>
 
-Simply run `dotnet pack --configuration Release` against the analyzer project to get a nuget package and publish it with
+</Project>
+```
 
-```shell
-dotnet nuget push {NugetPackageFullPath} -s nuget.org -k {NugetApiKey}
+Finally, add the following custom target to the [Directory.Build.targets](https://learn.microsoft.com/en-us/visualstudio/msbuild/customize-by-directory?view=vs-2022) file. This takes care of invoking analysis of single projects:
+```xml
+<Project>
+
+    <Target
+        Name="AnalyzeProject" 
+        DependsOnTargets="Restore;ResolveAssemblyReferencesDesignTime;ResolveProjectReferencesDesignTime;ResolvePackageDependenciesDesignTime;FindReferenceAssembliesForReferences;_GenerateCompileDependencyCache;_ComputeNonExistentFileProperty;BeforeBuild;BeforeCompile;CoreCompile">
+        
+        <Message Importance="normal" Text="fsc arguments: @(FscCommandLineArgs)" />
+        <Message Importance="High" Text="Analyzing $(MSBuildProjectFile)"/>
+        <Exec
+            ContinueOnError="true"
+            Command="dotnet fsharp-analyzers --fsc-args &quot;@(FscCommandLineArgs)&quot; --analyzers-path &quot;$(PkgG-Research_FSharp_Analyzers)\analyzers\dotnet\fs&quot; --exclude-analyzer PartialAppAnalyzer --fail-on-warnings GRA-STRING-001 GRA-STRING-002 GRA-STRING-003 GRA-UNIONCASE-001 --verbose --report &quot;$(MSBuildProjectName)-analysis.sarif&quot;">
+            <Output TaskParameter="ExitCode" PropertyName="LastExitCode" />
+        </Exec>
+        <Error Condition="'$(LastExitCode)' == '-2'" Text="Problems were found $(MSBuildProjectFile)" />
+
+    </Target>
+
+</Project>
 ```
 
-### Third-party dependencies
+You may need to adjust the `Command` to be compatible with your specific analyzer. Think about how you want warnings to be treated.
+
+⚠️ Note: This uses a feature from .NET 8 to get hold of the `FscCommandLineArgs`.
```

Comment:
No, this will work for non-dotnet 8 rc 2 projects as well.
The new feature is the ability to use `--getProperty` and `--getItem` from the command line.

Here you are essentially invoking a bunch of targets to do a design time build.

I think we probably want to have a short section on project cracking:

The way the analyzers work is that we will programmatically type-check a project and process the results with our analyzers. In order to do this programmatical type-check we need to construct `FSharpProjectOptions`. This is essentially a type that represents all the fsharp compiler arguments. When using `--project`, we will use [ProjInfo](https://www.nuget.org/packages/Ionide.ProjInfo) to invoke a set of MSBuild targets in the project to perform a design-time build. A design-time build is basically an empty invocation of a build. It won't produce a binary but will have constructed the correct arguments to theoretically invoke the compiler.

When we use `--fsc-args` we do exactly the same thing, but we tell MSBuild to invoke the targets. This happens in the slew of `BeforeTargets` and the properties we specify in the `Directory.Solution.targets` (`Properties="DesignTimeBuild=True;Configuration=Release;ProvideCommandLineArgs=True;SkipCompilerExecution=True"`). It is this exact combination that will essentially do the same thing as `--project`. The interesting thing about doing it via MSBuild is that some targets will be cached and this speeds up everything.

I hope this makes a little sense.